pkg/reconciler/common/prune.go

```diff
@@ -279,22 +283,74 @@ func (pruner *Pruner) checkAndCreate(ctx context.Context, uniquePruneNs, schedul
 				return err
 			}
 		}
-		return createCronJob(ctx, pruner.kc, suffixedCronName, pruner.targetNamespace, uniquePruneNs, schedule, pruneContainers, pruner.ownerRef)
+		return createCronJob(ctx, pruner.kc, suffixedCronName, pruner.targetNamespace, uniquePruneNs, schedule, pruneContainers, pruner.ownerRef, tC, computeTcConfigHash)
 	}
 
 	// no change in config but the cronjob does not exist
 	if len(cronList.Items) == 0 && !configChanged {
-		return createCronJob(ctx, pruner.kc, suffixedCronName, pruner.targetNamespace, uniquePruneNs, schedule, pruneContainers, pruner.ownerRef)
+		return createCronJob(ctx, pruner.kc, suffixedCronName, pruner.targetNamespace, uniquePruneNs, schedule, pruneContainers, pruner.ownerRef, tC, computeTcConfigHash)
 	}
 
 	// any case with config change
 	if configChanged {
-		return createCronJob(ctx, pruner.kc, suffixedCronName, pruner.targetNamespace, uniquePruneNs, schedule, pruneContainers, pruner.ownerRef)
+		return createCronJob(ctx, pruner.kc, suffixedCronName, pruner.targetNamespace, uniquePruneNs, schedule, pruneContainers, pruner.ownerRef, tC, computeTcConfigHash)
+	}
+	// get the cronjob to check previous annotations
+	cronjob := cronList.Items[0]
+	cronjobName := cronjob.Name
+	annotations := cronjob.Annotations
+	tektonConfigChanged := checkTektonConfigHashChange(annotations, computeTcConfigHash)
+
+	// if tektonConfig changes tC.Spec.Config
+	if tektonConfigChanged {
```

Comment:
How about changing `tektonConfigChanged`  to `nodeSelectorChanged` or something else.

Reply:
changed